pcdsdevices/epics/inout.py

```diff
@@ -1,64 +1,29 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-from ophyd import Device, Component as C
+from enum import Enum
 
-from .state import InOutStates, statesrecord_class
+from ..state import StatePositioner
 
 
-class InOutDevice(Device):
-    """
-    Device that has two states, IN and OUT, that blocks the beam while IN.
-    """
-    state = C(InOutStates, '')
-    #Subscription types
-    SUB_STATE = 'sub_state_changed'
-    _default_sub = SUB_STATE
-
-    def __init__(self, *args, **kwargs):
-        self._has_subscribed = False
-        super().__init__(*args, **kwargs)
+class InOutPositioner(StatePositioner):
+    _states_enum = Enum('InOutState', 'IN OUT')
 
     @property
     def inserted(self):
-        return self.state.value == 'IN'
+        return self.position == 'IN'
 
     @property
     def removed(self):
-        return self.state.value == 'OUT'
+        return self.position == 'OUT'
 
     def remove(self, wait=False, timeout=None, finished_cb=None, **kwargs):
-        return self.state.move('OUT', moved_cb=finished_cb, timeout=timeout,
-                               wait=wait, **kwargs)
+        return self.move('OUT', moved_cb=finished_cb, timeout=timeout,
+                         wait=wait, **kwargs)
 
-    def subscribe(self, cb, event_type=None, run=False):
-        """
-        Subscribe to changes of the InOutDevice
 
-        Parameters
-        ----------
-        cb : callable
-            Callback to be run
-
-        event_type : str, optional
-            Type of event to run callback on
-
-        run : bool, optional
-            Run the callback immediatelly
-        """
-        if not self._has_subscribed:
-            self.state.subscribe(self._on_state_change, run=False)
-            self._has_subscribed = True
-        super().subscribe(cb, event_type=event_type, run=run)
+class Diode(InOutPositioner):
+    pass
```

Comment:
What is this here for?

Reply:
There may be some desire to include a data readback on `Diode` in the future. I can remove this if you disagree.

Reply:
Up to you. More of a curiosity, but I am tempted to say delete it until we have a use case. Don't feel particularly strongly

Reply:
You're probably right, simpler is better. It's bad form to prepare for something that never happens.